Qq/Commands.lean

```diff
@@ -0,0 +1,98 @@
+import Qq.Macro
+import Lean
+
+namespace Qq
+open Lean Meta Elab Tactic
+
+private
+def mkHaveFVars (assignments : Array Expr) (body : Expr) : MetaM Expr := do
+  let ctx ← getLCtx
+  let (_, res) ← ctx.foldrM (init := (assignments, body)) fun decl (assignments, body) => do
+    let value := assignments.back!
+    let result := .letE decl.userName decl.type value (body.abstract #[.fvar decl.fvarId]) false
```

Comment:
Wouldn't a `have` be using `lambda` rather than `let`?

Reply:
I see, I was wondering if there is a way do all the abstraction at once, which feels more efficient than one by one.

Reply:
This doesn't work
```
  let ctx ← getLCtx
  let ctxLet := ctx.foldl (init := LocalContext.empty) (fun part decl =>
    part.addDecl (.ldecl decl.index decl.fvarId decl.userName
      decl.type assignments[decl.index]! false decl.kind))
  withLCtx ctxLet #[] do
    mkLetFVars assignments body
```